omni-relayer/src/workers/evm.rs

```diff
@@ -33,32 +41,117 @@ pub async fn finalize_withdraw(
 
         let mut handlers = Vec::new();
         for (key, event) in events {
-            if let Ok(withdraw_log) = serde_json::from_str::<Vec<u8>>(&event) {
+            if let Ok((block_number, log, tx_logs)) =
+                serde_json::from_str::<(u64, Log, Option<TransactionReceipt>)>(&event)
+            {
+                let Ok(light_client_latest_block_number) =
+                    utils::near::get_eth_light_client_last_block_number(&config, &jsonrpc_client)
+                        .await
+                else {
+                    warn!("Failed to get eth light client last block number");
+                    continue;
+                };
+
+                if block_number > light_client_latest_block_number {
```

Comment:
I assume that the events are in chronological order. Then it makes sense to add a sleep and jump to the loop.

Reply:
Fixed in 68821cd